wagtail/images/models.py

```diff
@@ -597,7 +598,10 @@ def get_cache_key(self, image):
 class AbstractRendition(ImageFileMixin, models.Model):
     filter_spec = models.CharField(max_length=255, db_index=True)
     file = models.ImageField(
-        upload_to=get_rendition_upload_to, width_field="width", height_field="height"
+        upload_to=get_rendition_upload_to,
+        storage=get_rendition_storage,
+        width_field="width",
+        height_field="height",
```

Comment:
as noted above, the function should refer to an instance method on AbstractRendition

```python
#  put something like this under `def get_rendition_upload_to(instance, filename):` ~line 65
def get_rendition_storage(instance):
    """
    Obtain a valid storage location for an image rendition file.

    This needs to be a module-level function so that it can be referenced within migrations,
    but simply delegates to the `get_storage` method of the instance, so that AbstractRendition
    subclasses can override it.
    """
    return instance.get_storage(filename)
```

```python
class AbstractRendition(ImageFileMixin, models.Model):
    # ...

    def get_upload_to(self, filename):
        # ...

    def get_storage(self):
        """
        Obtain the storage object for an image rendition file.
        Returns custom storage (if defined), or the default storage.
        """
        storage = getattr(settings, "WAGTAILIMAGES_RENDITION_STORAGE", default_storage)
        if isinstance(storage, str):
            import_module(storage)
            storage = storage()
        return storage
```